cmd/cluster/cpd.go

```diff
@@ -0,0 +1,188 @@
+package cluster
+
+import (
+	"fmt"
+
+	awsSdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/openshift/osdctl/pkg/osdCloud"
+	"github.com/openshift/osdctl/pkg/provider/aws"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+type cpdOptions struct {
+	clusterID  string
+	awsProfile string
+}
+
+const (
+	unknownProvisionCode = "OCM3999"
+)
+
+func newCmdCpd() *cobra.Command {
+	ops := cpdOptions{}
+	cpdCmd := &cobra.Command{
+		Use:               "cpd",
+		Short:             "Runs diagnostic for a Cluster Provisioning Delay (CPD)",
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	cpdCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", ops.clusterID, "Cluster ID")
+
+	return cpdCmd
+}
+
+func (o *cpdOptions) run() error {
+
+	// Get the cluster info
+	ocmClient := utils.CreateConnection()
+
+	resp, err := ocmClient.ClustersMgmt().V1().Clusters().Cluster(o.clusterID).Get().Send()
+	if err != nil {
+		return err
+	}
+
+	clusterInfo := resp.Body()
+
+	// Check if DNS is ready, exit out if not
+	if !clusterInfo.Status().DNSReady() {
+		fmt.Println("DNS not ready. Investigate reasons using the dnszones CR in the cluster namespace:")
+		fmt.Printf("oc get dnszones -n uhc-production-%s -o yaml --as backplane-cluster-admin\n", o.clusterID)
+		return nil
+	}
+
+	// Check if the OCM Error code is a known error
+	if clusterInfo.Status().ProvisionErrorCode() != unknownProvisionCode {
+		fmt.Printf("Error code %s is known, customer already received Service Log\n", clusterInfo.Status().ProvisionErrorCode())
+	}
+
+	// If the cluster is GCP, give instructions on how to get console access
+	if clusterInfo.CloudProvider().ID() == "gcp" {
+
+		return nil
```

Comment:
Should there be a "Not Implemented Yet" message here, at least?

Reply:
Ah yes, I forgot to do that. Will add